src/lib.rs

```diff
@@ -0,0 +1,37 @@
+#[deny(missing_docs)]
+
+/**
+* `lychee` is a library for checking links.
+* The main struct of this crate is `ClientBuilder` which can be used to
+* configure and run your own link checker.
+*
+* "Hello world" example:
+* ```
+*
+* use lychee::{ClientBuilder, Status};
+* use lychee::Uri::Website;
+* use url::Url;
+* use std::error::Error;
+*
+* #[tokio::main]
+* async fn main() -> Result<(), Box<dyn Error>> {
+*   let client = ClientBuilder::default().build()?;
+*   let url = Url::parse("https://github.com/lycheeverse/lychee")?;
+*   let response = client.check(Website(url)).await;
+*   assert!(matches!(response.status, Status::Ok(_)));
+*   Ok(())
+* }
+* ```
```

Comment:
Just sanity checking: is this actually being run as a doctest?

Reply:
Yup, and it passes. 😄

Reply:
```
   Doc-tests lychee

running 1 test
test src/lib.rs - client (line 8) ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

from https://github.com/lycheeverse/lychee/pull/46/checks?check_run_id=1471385577#step:4:674

Reply:
As a side-note, I'd really like to support 

```rust
 let response = client.check("https://github.com/lycheeverse/lychee").await;
```

instead of

```rust
*   let url = Url::parse("https://github.com/lycheeverse/lychee")?;
*   let response = client.check(Website(url)).await;
```

to simplify library usage. 
Of course it's not that easy because otherwise `check()` would have to return a `Result` and I don't like that. (The URL parsing could fail). Alternatively we return a `Response::Error` which is a special enum variant that indicates that parsing failed - but then we treat `Response` as some sort of `Result` which isn't very clean either.
The last option I can see is to go with the slightly less ergonomic

```rust
 let response = client.check("https://github.com/lycheeverse/lychee".try_into()?).await;
```

or 

```rust
 let response = client.check(Request::try_from("https://github.com/lycheeverse/lychee")?).await;
```

Not sure if it's worth the trouble, though. It would at least allow us to accept different input types in general, which could make it extensible after 1.0.
Did I miss an option? Feedback welcome.

Reply:
Very interesting ideas! Maybe we could provide 2 interfaces: one simplified e.g. "just pass as &str and I will tell you if it worked", and the "usual" one with explicit type handling.

But let's move this discussion to a dedicated place.

Reply:
Agreed. #61 is where it's at.